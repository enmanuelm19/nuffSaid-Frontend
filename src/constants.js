export const PRIORITIES = [
  { type: "error", title: "Error Type 1", cod: 1, style: "#F56236" },
  { type: "warning", title: "Warning Type 2", cod: 2, style: "#FCE788" },
  { type: "info", title: "Info Type 3", cod: 3, style: "#88FCA3" },
];

export const ERROR_CODE = 1;
