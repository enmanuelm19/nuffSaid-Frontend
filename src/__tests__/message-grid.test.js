import React from "react";
import { mount } from "enzyme";
import MessageGrid from "../components/message-grid";
import { create } from "react-test-renderer";
import { PRIORITIES } from "../constants";

describe("MessageGrid", () => {
  const messages = [
    { message: "First message", priority: 1 },
    { message: "Second message", priority: 2 },
    { message: "Third message", priority: 3 },
  ];

  const component = mount(<MessageGrid message={messages[0]} />);

  test("Render component", () => {
    expect(component.length).toEqual(1);
  });

  test("Render new message", () => {
    component.setProps({ message: messages[1] });
    const grid = create(component);
    expect(grid.toJSON()).toMatchSnapshot();
  });

  test("Render priorities lists", () => {
    expect(component.getDOMNode().childElementCount).toEqual(PRIORITIES.length);
  });
});
