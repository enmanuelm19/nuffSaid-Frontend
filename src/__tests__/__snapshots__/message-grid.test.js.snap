// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageGrid Render new message 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h2
          className="MuiTypography-root MuiTypography-h5"
        >
          Error Type 1
        </h2>
        <h4
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          Count 
          0
        </h4>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h2
          className="MuiTypography-root MuiTypography-h5"
        >
          Warning Type 2
        </h2>
        <h4
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          Count 
          0
        </h4>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h2
          className="MuiTypography-root MuiTypography-h5"
        >
          Info Type 3
        </h2>
        <h4
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          Count 
          0
        </h4>
      </div>
    </div>
  </div>
</div>
`;
